File: go-version/leetcode/LC1742/count_balls.go
package lc1742

func countBalls(lowLimit int, highLimit int) int {
	mp := make(map[int]int)
	mx := 0
	for ballNo := lowLimit; ballNo <= highLimit; ballNo++ {
		curBall, boxNo := ballNo, 0
		for curBall > 0 {
			boxNo += curBall % 10
			curBall /= 10
		}
		mp[boxNo]++
		if mp[boxNo] > mx {
			mx = mp[boxNo]
		}
	}
	return mx
}

File: python-version/LC1742/count_balls.py
class Solution:
    def countBalls(self, lowLimit: int, highLimit: int) -> int:
        mp = {}
        mx = 0
        for ballNo in range(lowLimit, highLimit+1):
            curBall, boxNo = ballNo, 0
            while curBall > 0:
                boxNo += curBall % 10
                curBall //= 10
            cnt = mp.get(boxNo, 0)+1
            mp[boxNo] = cnt
            mx = max(mx, cnt)
        return mx